internal/clients/http/gcp/gcp_client.go

```diff
@@ -91,6 +91,54 @@ func (c *gcpClient) newInstancesClient(ctx context.Context) (*compute.InstancesC
 	return client, nil
 }
 
+func (c *gcpClient) NewInstanceTemplatesClient(ctx context.Context) (*compute.InstanceTemplatesClient, error) {
+	client, err := compute.NewInstanceTemplatesRESTClient(ctx, c.options...)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create GCP templates client: %w", err)
+	}
+	return client, nil
+}
+
+func (c *gcpClient) ListLaunchTemplates(ctx context.Context) ([]*clients.LaunchTemplate, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "ListLaunchTemplates")
+	defer span.End()
+
+	logger := logger(ctx)
+	logger.Trace().Msgf("Listing launch templates")
```

Comment:
These "a function was called" are not very useful, in fact, thanks to opentelemetry exactly this message will be written in log (lines 103-104). But its fine.